src/linker/Linker.Steps/LinkAttributesParser.cs

```diff
@@ -154,26 +205,39 @@ static string FormatCustomAttribute (CustomAttribute ca)
 		{
 			var methods = attributeType.Methods;
 			for (int i = 0; i < attributeType.Methods.Count; ++i) {
-				var m = methods[i];
-				if (!m.IsInstanceConstructor ())
+				var method = methods[i];
+				if (!method.IsInstanceConstructor ())
 					continue;
 
-				var p = m.Parameters;
-				if (args.Length != p.Count)
+				var parameters = method.Parameters;
+				if (args.Length != parameters.Count
+					&& !(parameters.Count > 0
+						&& method.Parameters.Last ().CustomAttributes
+							.Any (ca => _context.Resolve (ca.AttributeType) == ParamArrayAttributeType)))
 					continue;
 
 				bool match = true;
-				for (int ii = 0; match && ii != args.Length; ++ii) {
+				for (int ii = 0; match && ii < args.Length; ++ii) {
```

Comment:
Why do we need this code here - I thought we are always using the object[] constructor now.

Reply:
If I understand what you're pointing to, this code finds the matching constructor for any attribute added by the XML, so it needs to account for more than `RemoveAttributeInstances`